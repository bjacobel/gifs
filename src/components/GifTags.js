import React, { Component, PropTypes } from 'react';

import Tag from './Tag';

export default class GifTags extends Component {
  render() {
    const { tags } = this.props;

    return (
      <p className="tag-list">
        { tags.map((tag) => {
          return (
            <Tag key={ tag.id } content={ tag.text } />
          );
        }) }
        <Tag meta="add-tag" />
      </p>
    );
  }
}

GifTags.propTypes = {
  tags: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
};
