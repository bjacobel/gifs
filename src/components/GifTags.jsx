import React, { Component, PropTypes } from 'react';

export default class GifTags extends Component {
  render() {
    const { tags } = this.props;

    return (
      <div>
        { tags.map((tag) => {
          return (
            <span key={ tag.id }>{ tag.text }</span>
          );
        }) }
      </div>
    );
  }
}

GifTags.propTypes = {
  tags: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
};
