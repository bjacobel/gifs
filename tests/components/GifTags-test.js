import React from 'react';
import { shallow } from 'enzyme';

import GifTags from '../../src/components/GifTags';

describe('GifTags component', () => {
  it('matches snapshot with no tags passed', () => {
    expect(shallow(<GifTags tags={ [] } />)).toMatchSnapshot();
  });

  describe('when passed tags', () => {
    const tags = [
      { id: '1', text: '1' },
      { id: '2', text: '2' },
      { id: '3', text: '3' },
      { id: '4', text: '4' },
      { id: '5', text: '5' },
    ];

    it('renders as many <Tag> components as tags passed', () => {
      const enzymeRepr = shallow(<GifTags tags={ tags } />);
      expect(enzymeRepr.find('Connect(Tag)').length).toEqual(tags.length + 1);
      expect(enzymeRepr.find('Connect(Tag)').last().prop('meta')).toEqual('add-tag');
    });

    it('matches snapshot', () => {
      const enzymeRepr = shallow(<GifTags tags={ tags } />);
      expect(enzymeRepr).toMatchSnapshot();
    });
  });
});
