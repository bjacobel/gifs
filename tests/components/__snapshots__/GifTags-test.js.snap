exports[`GifTags component matches snapshot with no tags passed 1`] = `
<p
  className="tag-list">
  <Connect(Tag)
    meta="add-tag" />
</p>
`;

exports[`GifTags component when passed tags matches snapshot 1`] = `
<p
  className="tag-list">
  <Connect(Tag)
    content="1" />
  <Connect(Tag)
    content="2" />
  <Connect(Tag)
    content="3" />
  <Connect(Tag)
    content="4" />
  <Connect(Tag)
    content="5" />
  <Connect(Tag)
    meta="add-tag" />
</p>
`;
